backend/hdf5/DimensionHDF5.cpp

```diff
@@ -259,13 +269,153 @@ void SetDimensionHDF5::labels(const none_t t) {
+    // this way will not have the block and file
+}
+
+DataFrameDimensionHDF5::DataFrameDimensionHDF5(const H5Group &group, ndsize_t index,
+                                               const std::shared_ptr<IFile> &file,
+                                               const std::shared_ptr<IBlock> &block)
+    : DimensionHDF5(group, index), entity_block(block), entity_file(file)
+{
+    setType();
+}
+
+DataFrameDimensionHDF5::DataFrameDimensionHDF5(const H5Group &group, ndsize_t index,
+                                               const std::shared_ptr<IFile> &file,
+                                               const std::shared_ptr<IBlock> &block, const DataFrame &frame)
+    :DataFrameDimensionHDF5(group, index, file, block)
+{
+    std::shared_ptr<IDataFrame> idf = block->getEntity<IDataFrame>(frame.id());
+    if (!idf)
+        throw std::runtime_error("ColumnDimensionHDF5 DataFrame not found in block!");
+    if (this->group.hasGroup("data_frame"))
+        this->group.removeGroup("data_frame");
+    auto target = std::dynamic_pointer_cast<DataFrameHDF5>(idf);
+
+    setType();
+    this->group.createLink(target->group(), "data_frame");
+}
+
+DataFrameDimensionHDF5::DataFrameDimensionHDF5(const H5Group &group, ndsize_t index,
+                                               const std::shared_ptr<IFile> &file,
+                                               const std::shared_ptr<IBlock> &block, const DataFrame &frame,
+                                               unsigned col_index)
+    :DataFrameDimensionHDF5(group, index, file, block, frame)
+{
+    this->group.setAttr("column_index", col_index);
+}
+
+int DataFrameDimensionHDF5::columnIndex() const {
+    int col_index = -1;
+    if (group.hasAttr("column_index")) {
+        group.getAttr("column_index", col_index);
+    }
+    return col_index;
+}
+
+int DataFrameDimensionHDF5::checkColumnIndex(int col_index) const {
+    int column_index = col_index;
+    if (col_index == -1) {
+        column_index = columnIndex();
+    }
+
+    if (column_index == -1) {
+        throw nix::OutOfBounds("DataFrameDimension: Error accessing column, no column index was given and no default is specified.");
+    }
+
+    nix::DataFrame df = dataFrame();
+    std::vector<Column> cols = df.columns();
+    if (static_cast<size_t>(column_index) >= cols.size()) {
+        throw nix::OutOfBounds("DataFrameDimension: Error accessing column, column index exceeds number of columns!");
+    }
+
+    return column_index;
+}
+
+
+DimensionType DataFrameDimensionHDF5::dimensionType() const {
+    return DimensionType::DataFrame;
+}
+
+
+std::string DataFrameDimensionHDF5::unit(int col_index) const {
+    int column_index = checkColumnIndex(col_index);
+
+    nix::DataFrame df = dataFrame();
+    std::vector<Column> cols = df.columns();
+    return cols[column_index].unit;
+}
+
+
+std::string DataFrameDimensionHDF5::label(int col_index) const {
```

Comment:
if `int` is used as a way to indicate that you might not actually want to select a column, maybe `boost::optional<unsigned>` would be better, so we don't have to mix types between signed and unsigned.